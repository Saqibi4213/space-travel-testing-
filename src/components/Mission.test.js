import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Missions from './Missions';
import '@testing-library/jest-dom';
import { join } from '../redux/missions/missionSlice';

const mockMissions = [
  {
    mission_id: '1',
    mission_name: 'Test Mission',
    description: 'Test Description',
    reserved: false,
  },
];

const middlewares = [];
const mockStore = configureMockStore(middlewares);

describe('Missions Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      missions: {
        missions: mockMissions,
        status: 'succeeded',
      },
    });
  });

  test('show missions', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(getByText('Not a Member')).toBeInTheDocument();
    expect(getByText('Join Mission')).toBeInTheDocument();
    expect(getByText('Test Mission')).toBeInTheDocument();
    expect(getByText('Test Description')).toBeInTheDocument();
  });

  test('join mission', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    const btnJoin = getByText('Join Mission');
    fireEvent.click(btnJoin);

    expect(store.getActions()).toContainEqual(join('1'));
  });
  test('all missions snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
