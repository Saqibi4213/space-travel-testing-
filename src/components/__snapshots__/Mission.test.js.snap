// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component all missions snapshot 1`] = `
<DocumentFragment>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th
          class="mission-name"
        >
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test Mission
        </td>
        <td>
          Test Description
        </td>
        <td
          class="wide-td"
        >
          <span
            class="member-span"
            style="color: white; background-color: rgb(119, 119, 119); border-radius: 3px; cursor: default; padding: 3px;"
          >
            Not a Member
          </span>
        </td>
        <td
          class="wide-td"
        >
          <button
            style="border: 1px solid black; background-color: transparent; color: black; padding: 5px; border-radius: 4px;"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
